google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/facet/impl/AppEngineStandardUltimateWebIntegration.java

```diff
@@ -74,10 +80,48 @@ public ArtifactType getAppEngineWebArtifactType() {
   public VirtualFile suggestParentDirectoryForAppEngineWebXml(
       @NotNull Module module, @NotNull ModifiableRootModel rootModel) {
     final WebFacet webFacet = ContainerUtil.getFirstItem(WebFacet.getInstances(module));
-    if (webFacet == null) {
+    if (webFacet != null) {
+      VirtualFile webXmlParent = getWebXmlParent(webFacet);
+      if (webXmlParent != null) {
+        return webXmlParent;
+      }
+
+      VirtualFile webInfPath = getWebInfPath(webFacet);
+      if (webInfPath != null) {
+        return webInfPath;
+      }
+    }
+
+    return super.suggestParentDirectoryForAppEngineWebXml(module, rootModel);
+  }
+
+  private VirtualFile getWebInfPath(@NotNull WebFacet webFacet) {
+    List<WebRoot> webRoots = webFacet.getWebRoots();
+    if (webRoots.isEmpty()) {
+      return null;
+    }
+
+    for (WebRoot webRoot : webRoots) {
+      VirtualFile parent = webRoot.getFile();
+      if (parent.getPath().endsWith("/WEB-INF")) {
```

Comment:
Could we use `parent.getName()` instead of string comparison for the whole path maybe? It usually supposed to return the name of the actual last file in the path.